cli/viam/main.go

```diff
@@ -905,6 +909,46 @@ viam module upload --version "0.1.0" --platform "linux/amd64" packaged-module.ta
 					},
 				},
 			},
+			{
+				Name:  "version",
+				Usage: "print version info for this program",
+				Flags: []cli.Flag{
+					&cli.BoolFlag{
+						Name:  "debug",
+						Usage: "dump full BuildInfo",
+					},
+				},
+				Action: func(c *cli.Context) error {
+					if info, ok := debug.ReadBuildInfo(); !ok {
+						log.Fatal("Error reading build info")
+					} else {
+						if c.Bool("debug") {
+							fmt.Fprintf(c.App.Writer, "%s\n", info.String())
+						}
+						settings := make(map[string]string, len(info.Settings))
+						for _, setting := range info.Settings {
+							settings[setting.Key] = setting.Value
+						}
+						version := "?"
+						if rev, ok := settings["vcs.revision"]; ok {
+							version = rev[:8]
+							if settings["vcs.modified"] == "true" {
+								version += "+"
+							}
+						}
+						deps := make(map[string]*debug.Module, len(info.Deps))
+						for _, dep := range info.Deps {
+							deps[dep.Path] = dep
+						}
+						apiVersion := "?"
+						if dep, ok := deps["go.viam.com/api"]; ok {
+							apiVersion = dep.Version
+						}
+						fmt.Fprintf(c.App.Writer, "version %s git=%s api=%s %s\n", CliVersion, version, apiVersion, info.GoVersion)
```

Comment:
What Go version is this printing? The version of Go used to build the binary? If not, it'd be a little odd to print the current user's version of Go when they run the `viam` binary locally.

Reply:
'version of the toolchain that built the binary' (per [this](https://pkg.go.dev/runtime/debug#BuildInfo))

honestly the edge case where this matters is if someone has their own build of the tool and used a version that is new enough to build, but has some bug that causes bad behavior. this is rare -- I'll remove the go version field

Reply:
Cool; SGTM.